### PR TITLE
Move molecule tests out of role

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Molecule tests should not be in a role, need to move them out of roles, and fix the paths.

This will help in the transition to collections

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Ran a few scenarios locally, jenkins job must pass

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: molecule/archive-community-plaintext-rhel/molecule.yml [hunk 1]
```diff
@@ -77,11 +77,13 @@ platforms:
   # Control center not included in community archive, must not have c3 in inventory
 provisioner:
   name: ansible
+  env:
```
Comment (domenicbove) [inline molecule/archive-community-plaintext-rhel/molecule.yml:80]: this env var might not be necessary with latest molecule

Comment (domenicbove) [reply molecule/archive-community-plaintext-rhel/molecule.yml:80]: I just confirmed this env var is not needed with my molecule version 
```
% molecule --version
molecule 3.3.0 using python 3.8
    ansible:2.11.1
    delegated:3.3.0 from molecule
    docker:0.2.4 from molecule_docker
```

I vaguely remember needing it in previous molecule version. Do you think we should have it/ be redundant or remove it and potentially force ppl to upgrade molecule?

Comment (erikgb) [reply molecule/archive-community-plaintext-rhel/molecule.yml:80]: If it is required in older versions of molecule, I think we can/should keep it. As long as it still works in newer versions.

File: HOW_TO_TEST.md [hunk 2]
```diff
@@ -58,7 +58,7 @@ Molecule allows for testing a role and will live inside a role's directory in a
 
 To get a list of the scenarios:
 
-```ls roles/confluent.test/molecule```
+```ls molecule```
```
Comment (erikgb) [inline HOW_TO_TEST.md:61]:
```suggestion
```molecule list```
```
Gives more information. Could also keep it as is....

Comment (domenicbove) [reply HOW_TO_TEST.md:61]: im not sure i like the output of molecule list it shows all the containers, its a little overkill

File: molecule/archive-community-plaintext-rhel/molecule.yml (hunk 1 above)
Comment (domenicbove) [reply molecule/archive-community-plaintext-rhel/molecule.yml:80]: Looks like jenkins is using molecule 3.0 and the env var is not required
```
12:36:31  + molecule --version
12:36:31  molecule 3.0.2.1
12:36:31     ansible==2.9.21 python==2.7
```
should remove that env var

Comment (domenicbove) [reply molecule/archive-community-plaintext-rhel/molecule.yml:80]: Correction, you need that env var for molecule 3.0, probably safest to keep it then:
```
[31mERROR! the role 'confluent.test.ldap' was not found in /home/jenkins/workspace/c_Contributors_cp-ansible_PR-701/molecule/rbac-scram-custom-rhel/../roles:/home/jenkins/.cache/molecule/c_Contributors_cp-ansible_PR-701/rbac-scram-custom-rhel/roles:/home/jenkins/workspace:/home/jenkins/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/home/jenkins/workspace/c_Contributors_cp-ansible_PR-701/molecule/rbac-scram-custom-rhel/..[0m
```

Comment (domenicbove) [reply molecule/archive-community-plaintext-rhel/molecule.yml:80]: Correction, you need that env var for molecule 3.0, probably safest to keep it then:
```
[31mERROR! the role 'confluent.test.ldap' was not found in /home/jenkins/workspace/c_Contributors_cp-ansible_PR-701/molecule/rbac-scram-custom-rhel/../roles:/home/jenkins/.cache/molecule/c_Contributors_cp-ansible_PR-701/rbac-scram-custom-rhel/roles:/home/jenkins/workspace:/home/jenkins/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/home/jenkins/workspace/c_Contributors_cp-ansible_PR-701/molecule/rbac-scram-custom-rhel/..[0m
```

Comment (domenicbove) [reply molecule/archive-community-plaintext-rhel/molecule.yml:80]: can consider for follow up work

**Review by erikgb — APPROVED** [review]
LGTM

**Comment by erikgb** [comment]
@domenicbove You should consider to squash the commits before/on merge.

**Pull request merged**